### DISCUSSION TITLE
Bug with Invoke-PnPSiteTemplate fails for libraries that have been renamed

### DISCUSSION THREAD

**Discussion by scallin** [pnp/powershell#1869] [Q&A]
### Discussed in https://github.com/pnp/powershell/discussions/1854

<div type='discussions-op-text'>

<sup>Originally posted by **scallin** May 11, 2022</sup>
Get-PnPSiteTemplate to create the site template xml, then use Invoke-PnPSiteTemplate to apply it to another site.  Invoke fails if any library has been renamed (Title is different from the URL).

The library was created as "Meetings" then renamed "Departmental Meetings 2022", the template xml looks correct:

![image](https://user-images.githubusercontent.com/20932971/167865633-ee452843-91f5-46fc-9a58-3a11c0b73042.png)

The invoke command fails because it's looking for the form in the right location (meetings/forms/dispform.aspx) but the library was actually re-created using the title Departmental Meetings 2022

![image](https://user-images.githubusercontent.com/20932971/167868285-53fa3380-e47f-4dff-bd62-5fc3ad8872e6.png)

The library is successfully recreated, but the URL for the new library is the Title, not the Url from the template

![image](https://user-images.githubusercontent.com/20932971/167866913-4428874c-07ce-4be5-a1c1-7406ff5eebac.png)
</div>

**Comment by veronicageek** [comment]
Please open an issue if necessary by **filling the template** so we have the maximum information.
